### ISSUE TITLE
Assertion `t->trace_reader().read_raw_data_for_frame_with_holes(buf)' failed to hold

### ISSUE THREAD

**Issue by sidkshatriya** [rr-debugger/rr#2861]
The `mremap_private_grow_under_map` test now fails with: 

```
[FATAL ../src/replay_syscall.cc:590:write_mapped_data()] 
 (task 514265 (rec:514262) at time 255)
 -> Assertion `t->trace_reader().read_raw_data_for_frame_with_holes(buf)' failed to hold. 
```

This is probably related to the recent mapping of sparse files changes. I'm using Ubuntu 20.04.2

I can post more information but its probably easier for you to try on 20.04 if you have it installed already.

**Comment by rocallahan** [comment]
Fixed by 1d826918ac3da223e652b2613e195d78f9c43a74

**Comment by sidkshatriya** [comment]
Yes, the problem looks fixed now. While using current master at e9ec388d95ae05bc2eceb0361aea39510f08e17e I don't encounter this problem anymore.
